pkg/nvpci/nvpci.go

```diff
@@ -428,3 +482,39 @@ func (p *nvpci) GetGPUByIndex(i int) (*NvidiaPCIDevice, error) {
 
 	return gpus[i], nil
 }
+
+func (p *nvpci) getSriovInfoForPhysicalFunction(devicePath string) (*SriovInfo, error) {
+	var sriovInfo *SriovInfo
+	sriovTotalVfs, err := os.ReadFile(filepath.Join(devicePath, "sriov_totalvfs"))
+	if err == nil {
+		totalVfsStr := strings.TrimSpace(string(sriovTotalVfs))
+		maxVfs, err := strconv.ParseUint(totalVfsStr, 10, 16)
+		if err != nil {
+			return nil, fmt.Errorf("unable to convert sriov_totalvfs to uint64: %v", err)
+		}
+		sriovInfo = &SriovInfo{
+			PhysicalFunction: &SriovPhysicalFunction{
+				TotalVFs: maxVfs,
+			},
+		}
+	} else if os.IsNotExist(err) { // not an SRIOV device return nil
+		return sriovInfo, nil
+	} else {
+		return nil, fmt.Errorf("unable to detect sriov_totalvfs: %v", err)
+	}
+
+	sriovNumVfs, err := os.ReadFile(filepath.Join(devicePath, "sriov_numvfs"))
+	if err != nil {
+		return nil, fmt.Errorf("unable to detect sriov_numvfs for: %v", err)
+	}
+
+	numVfsStr := strings.TrimSpace(string(sriovNumVfs))
+	vfs, err := strconv.ParseUint(numVfsStr, 10, 16)
+	if err != nil {
+		return nil, fmt.Errorf("unable to convert sriov_numvfs to uint64: %v", err)
+	}
+
+	sriovInfo.PhysicalFunction.NumVFs = vfs
+
+	return sriovInfo, nil
+}
```

Comment:
Some suggestions to improve readability...

```suggestion
func (p *nvpci) getSriovInfoForPhysicalFunction(devicePath string) (*SriovInfo, error) {
	totalVfsPath := filepath.Join(devicePath, "sriov_totalvfs")
	numVfsPath := filepath.Join(devicePath, "sriov_numvfs")

	// No file for sriov_totalvfs exists? Not an SRIOV device, return nil
	_, err := os.Stat(totalVfsPath)
	if err != nil && os.IsNotExist(err) {
		return nil, nil
	}

	sriovTotalVfs, err := os.ReadFile(filepath.Join(totalVfsPath)
	if err != nil { 
		fmt.Errorf("unable to read sriov_totalvfs: %v", err)
	}

	totalVfsStr := strings.TrimSpace(string(sriovTotalVfs))
	totalVfsInt, err := strconv.ParseUint(totalVfsStr, 10, 16)
	if err != nil {
		return nil, fmt.Errorf("unable to convert sriov_totalvfs to uint64: %v", err)
	}
	
	sriovNumVfs, err := os.ReadFile(numVfsPath)
	if err != nil {
		return nil, fmt.Errorf("unable to read sriov_numvfs for: %v", err)
	}

	numVfsStr := strings.TrimSpace(string(sriovNumVfs))
	numVfsInt, err := strconv.ParseUint(numVfsStr, 10, 16)
	if err != nil {
		return nil, fmt.Errorf("unable to convert sriov_numvfs to uint64: %v", err)
	}
	
	sriovInfo := &SriovInfo{
		PhysicalFunction: &SriovPhysicalFunction{
			TotalVFs: totalVfsInt,
			NumVFs: numVfsInt,			
		},
	}

	return sriovInfo, nil
}
```

Reply:
Agreed.